lib/declare_schema/model/table_options_definition.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module DeclareSchema
+  module Model
+    class TableOptionsDefinition
+      include Comparable
+
+      TABLE_OPTIONS_TO_SQL_MAPPINGS = {
+        charset:   'CHARACTER SET',
+        collation: 'COLLATE'
+      }.freeze
+
+      class << self
+        def for_model(model, old_table_name = nil)
+          table_name    = old_table_name || model.table_name
+          table_options = if model.connection.class.name.match?(/mysql/i)
+                            mysql_table_options(model.connection, table_name)
+                          else
+                            {}
+                          end
+
+          new(table_name, table_options)
+        end
+
+        private
+
+        def mysql_table_options(connection, table_name)
+          query = <<~EOS
+            SELECT CCSA.character_set_name, CCSA.collation_name
+            FROM information_schema.`TABLES` T, information_schema.`COLLATION_CHARACTER_SET_APPLICABILITY` CCSA
+            WHERE CCSA.collation_name = T.table_collation AND T.table_schema = "#{connection.current_database}" AND T.table_name = "#{table_name}";
+          EOS
+          defaults = connection.select_one(query)
+
+          {
+            charset:   defaults["character_set_name"],
+            collation: defaults["collation_name"]
+          }
```

Comment:
Can you document the contract for this method? Does it guarantee to have the keys `charset:` and `collation:`? Are those values guaranteed to be non-nil? If so, can you enforce that here?
```
          {
            charset:   defaults["character_set_name"] or raise "character_set_name missing from #{defaults.inspect}",
            collation: defaults["collation_name"] or raise "collation_name missing from #{defaults.inspect}",
          }
```

Reply:
They are guaranteed to be non-nil, so I've gone ahead and updated the contract as you suggested 851956a5c5a4c55ab99e69a9b461201df9c7251d